### PR TITLE
add filter functions

### PR DESCRIPTION
**Pull request by YehyaTamimi**

### CODE REVIEW

**Comment by DuaaH** [comment]
// Define an array of button classes instead calling each one separately
const buttonClasses = [".price-button", ".size-button", ".rooms-button"];
then Loop through each button class and add event listeners

File: Js/search.js [hunk 1]
```diff
@@ -72,57 +72,130 @@ const createPriceRangeElement = (element) => {
...
+                container.removeChild(card);
+            }
+        })
+    } else {
+        cards.forEach((card) => {
+            let size = card.querySelector(".small-info").textContent.split("|")[2].split(" ")[1];
+            if (!(size >= min && size <= max)) {
+                container.removeChild(card);
+            }
+        })
```
Comment (DuaaH) [inline Js/search.js:112]: create a new value then use it depend on condition and remove the duplication

File: Js/search.js [hunk 2]
```diff
@@ -72,57 +62,123 @@ const createPriceRangeElement = (element) => {
...
+            let price = card.querySelector(".price").textContent.replace(/\,/g, '').split('$')[1];
+            if (!(price >= min && price <= max)) {
+                container.removeChild(card);
+            }
+        } else {
+            let size = card.querySelector(".small-info").textContent.split("|")[2].split(" ")[1];
+            if (!(size >= min && size <= max)) {
+                container.removeChild(card);
+            }
+        }
```
Comment (DuaaH) [inline Js/search.js:100]: still need to refactor this 
```
let value;
        if (type === "price") {
            value = card.querySelector(".price").textContent.replace(/\,/g, '').split('$')[1];
        } else {
            value = card.querySelector(".small-info").textContent.split("|")[2].split(" ")[1]
        }

```        
        and add the other code

Comment (YehyaTamimi) [reply Js/search.js]: fixed

**Comment by DuaaH** [comment]
please fix this 
image

**Comment by YehyaTamimi** [comment]
> please fix this image

i made it so that it shows any houses that have an equal number or more than to that in the applied filter, should i make it show house that have the exact number applied by the filter only?

**Comment by DuaaH** [comment]
> > please fix this image
> 
> i made it so that it shows any houses that have an equal number or more than to that in the applied filter, should i make it show house that have the exact number applied by the filter only?

yes, please

**Comment by DuaaH** [comment]
please keep the selected number highlited
image

**Comment by YehyaTamimi** [comment]
> please keep the selected number highlited image

implemented in the history branch, implement it here?

**Comment by DuaaH** [comment]
> > please keep the selected number highlited image
> 
> implemented in the history branch, implement it here?

no need, I will check it in the history branch

**Comment by YehyaTamimi** [comment]
> > > please keep the selected number highlited image
> > 
> > 
> > implemented in the history branch, implement it here?
> 
> no need, I will check it in the history branch

ok, i will merge this PR now

**Pull request merged**